include/libcyphal/transport/udp/udp_transport_impl.hpp

```diff
@@ -717,13 +768,268 @@ class TransportImpl final : private TransportDelegate, public IUdpTransport  //
         return cetl::nullopt;
     }
 
+    /// @brief Tries to call an action with a media and its RX socket.
+    ///
+    /// The RX socket is made on demand if necessary (but `endpoint` parameter should have a value).
```

Comment:
Shouldn't we only need to make the RX socket once during initialization? Is this essentially creating a new RX socket every time we call receive?

Reply:
These are made only when there is no one previously created. Such empty condition (namely UniquePtr is equal to`nullptr`) may happen in two scenarios:
- the very first call when RX subscription/service is made (see `makeMsgRxSession` & `makeSvcRxSession`)
- on transport `run` call - when for any reason the first (above) scenario had failed (but the transient error handler said "it's fine, continue!"), user have got a valid RX session, and now this session is part of transport's `run` life - hence transport tries to "recover" from this empty socket condition but attempting again. Implementation of `IMedia` may do different strategies in such case. For example:
a. do nothing, let it be, and very quickly return error again - has relatively small impact
b. really try create a socket again and again on each run cycle until it succeeded eventually.
c. have some kind of "max attempts counter" and do the "b" several times, and then switch to "a"
d. or, make it time based - like occasionally, once in a while, do retry recreate such faulty media socket, but not on every attempt.

Note, that currently we don't have means to "mark" a media as permanently "broken", so yes, at transport level we do retry recreate socket on each run, but once it succeeded we of course stop doing it (b/c we finally got one).

Reply:
Perfect, thank you for the description!